### PR TITLE
Smallpeice 2023

### LINKED ISSUE #18: Clarify dependencies
On Ubuntu I found that I also needed `texlive-latex-extra` and `texlive-science`.

### LINKED ISSUE #19: Newer(?) inkscape uses --export-pdf not --export-filename

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Write the rules for smallpeice 2023.

Fixes #18 
Fixes #19 

This also ports the tooling from LaTeX to HTML, as it's much easier to work with, especially cross-platform. I've gone with `vite` as a build tool to simplify building the final HTML with styles, such that it can be easily deployed to GitHub Pages.

The template is massively "borrowed" from the Student Robotics rulebook, but updated to be Smallpeice / SourceBots specific.

<details><summary>Render</summary>
<p>

image

</p>
</details>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+          <tbody>
+            <tr>
+              <td>Arena boundary</td>
+              <td>0 - 27</td>
+              <td>200</td>
+            </tr>
+          </tbody>
+        </table>
+        <p>
+          Two sets of marker codes will be used: one for development purposes, and one for the competition itself. The competition set is only to be used inside the SourceBots arena at the SourceBots competition. This is so that people carrying markers past the arena do not confuse robots. The competition codes are 100 above the development codes. When run in competition mode, the software provided by SourceBots will subtract 100 from the detected marker codes, as well as ignore the development codes. Markers outside of the ranges used in the game will also be ignored.
```
Comment (RealOrangeOne) [inline index.html:154]: Question: Do we use competition markers at smallpeice? Previous rules don't mention it, so I presume not?

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          Flags are not counted when considering the starting size of the robot.
+        <li>
+          The identifying part of the robot flag must be visible when attached to the mount.
+        <li>
+          The mounting cylinder must be permanently affixed to the main chassis of a robot, and vertical when the robot is in its typical stopped position.
+      </ol>
+      <h3><a class="anchor" id="can-specification"></a>Cans</h3>
+      <ol>
+        <li>
+          Cans are cylindrical metallic containers weighing 300g ±20g.
```
Comment (WillB97) [inline src/index.html:200]: You need to account for the metal of the can
```suggestion
          Cans are cylindrical metallic containers weighing 350g ±20g.
```

File: index.html (hunk 1 above)
Comment (WillB97) [reply index.html]: No, sbot doesn't seen to have any marker offset support.

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        <li>
+          Any assistance from Smallpeice Trust staff and volunteers is provided without guarantees.
+        <li>
+          No remote control systems or radio transmitters or receivers are permitted.
+        <li>
+          While accidental bumps and scrapes are inevitable, the sport is non-contact.
+        <li>
+          Robots must pass a safety inspection before they can enter a physical arena.
+        <li>
+          Robots must not deliberately or negligently damage the arena or anything in it.
```
Comment (sedders123) [inline src/index.html:96]: See L100
```suggestion
```

**Review by sedders123 — COMMENTED** [review]
Looks good, a few minor things

File: src/index.html [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+            <td>0 - 7</td>
+            <td>100</td>
+          </tr>
+        </tbody>
+      </table>
+      <p>
+        The markers can be printed on a black-and-white printer.
+      </p>
+      <p>
+        Eight arena wall markers are positioned around the arena as shown in the <a href="#arena-specification">arena specification</a>. Marker codes are mirrored on each side of the arena.
```
Comment (sedders123) [inline src/index.html:159]:
```suggestion
        Sixteen arena wall markers are positioned around the arena as shown in the <a href="#arena-specification">arena specification</a>. Marker codes are mirrored on each side of the arena.
```

File: src/index.html [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        <li>
+          Each match lasts for <strong>150</strong> seconds.
+        <li>
+          Robots will be started by, or at the direction of, match officials.
+        <li>
+          A point is awarded each time a robot crosses a scoring line in the anti-clockwise direction.
+        <li>
+          Robots can pick up tin cans which are in the track. For every can a robot is carrying when it crosses a scoring line, the number of points awarded is doubled.<br>For example, a robot carrying two cans is awarded four points.
+        <li>
+          At the end of each lap - defined as every sixth crossing of a scoring line in a forward direction - a robot is awarded an additional four points above the points it would normally get for the crossing.
```
Comment (prophile) [inline src/index.html:62]: Could we move the parenthetical to a footnote maybe?

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+        <li>
+          At the end of a match, each robot is awarded <strong>two</strong> additional points for each tin can it is carrying.
```
Comment (prophile) [inline src/index.html:64]: Particularly on paper I think these might be better spelled out in digits

File: src/index.html [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+        <li>
+          A robot is deemed to have passed a scoring line when the back of the robot passes the line.
+        <li>
+          Cases of a robot passing backwards over a line are offset against future crossings forward of a line. That is, if a robot crosses over two track boundaries backwards, it will need to cross two track boundaries forwards before it can gain any more scoring line points.
```
Comment (prophile) [inline src/index.html:68]: This is a little bit wordy, would be good if we can optimise this down a bit?

File: src/index.html [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+        <li>
+          Teams must conduct themselves within the principles of good sporting conduct.
+        <li>
+          Any assistance from Smallpeice Trust staff and volunteers is provided without guarantees.
+        <li>
+          No remote control systems or radio transmitters or receivers are permitted.
+        <li>
+          While accidental bumps and scrapes are inevitable, the sport is non-contact.
+        <li>
+          Robots must pass a safety inspection before they can enter a physical arena.
```
Comment (prophile) [inline src/index.html:94]: I think "before they can compete" might be better

File: src/fig-arena.svg [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+       class="legend-label"
+       y="2.2"
+       x="1.2">Can</text>
+
+    <use xlink:href="#marker" transform="translate(0.5 3.4) scale(2) rotate(90)" class="marker"/>
+    <text class="marker-label" y="3.4" x="0.7">2</text>
+    <text
+       class="legend-label"
+       y="3.4"
+       x="1.2">Marker (with id)</text>
```
Comment (prophile) [inline src/fig-arena.svg:309]: "ID" uppercase I think, otherwise it looks a bit Freud

**Review by prophile — APPROVED** [review]

File: src/index.html (hunk 7 above)
Comment (RealOrangeOne) [reply src/index.html:68]: Perhaps we just remove the example?

```suggestion
          Cases of a robot passing backwards over a line are offset against future crossings forward of a line.
```

**Pull request merged**